sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/JavaTypeInferenceSuite.scala

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst
+
+import java.math.BigInteger
+
+import scala.beans.BeanProperty
+
+import org.apache.spark.SparkFunSuite
+import org.apache.spark.sql.catalyst.expressions.{CheckOverflow, Expression, Literal}
+import org.apache.spark.sql.types.DecimalType
+
+class DummyBean(@BeanProperty var bigInteger: BigInteger) {
+}
+
+class JavaTypeInferenceSuite extends SparkFunSuite {
```

Comment:
Although we don't have this test suite, is this inevitable? It seems like an overkill for a single simple test case.

Reply:
where do I suggest you put the test case?

Reply:
I guess we don't have a test suite for this. I think it's not terrible to add it; ideally we'd also add other simple tests here. Or skip it. I'm fine with any.

Reply:
@dongjoon-hyun  @srowen  Anything else I can do here? I prefer to keep the JavaTypeInferenceSuite  here and add tests over time to it.

Reply:
I'll merge shortly if there are no objections. I think it's OK as an improvement, don't see any downside

Reply:
+1 for @srowen 's comment.